tango_ros_common/tango_ros_native/src/tango_ros_node.cpp

```diff
@@ -974,67 +983,82 @@ void TangoRosNode::PublishMeshMarker() {
+          if(result != TANGO_3DR_SUCCESS) {
+            LOG(ERROR) << "Tango3DR_extractMeshSegment failed.";
+            continue;
+          }
+          if (tango_mesh.num_faces == 0) {
+            LOG(INFO) << "Empty mesh extracted.";
+            continue;
+          }
+          // Make mesh marker from tango mesh.
+          visualization_msgs::Marker mesh_marker;
+          tango_ros_conversions_helper::toMeshMarker(t3dr_updated_indices.indices[i],
+                                             tango_mesh, time_offset_, &mesh_marker);
+          // Free tango mesh once we are finished with it.
+          result = Tango3DR_Mesh_destroy(&tango_mesh);
+          if (result != TANGO_3DR_SUCCESS) {
+            LOG(ERROR) << "Tango3DR_Mesh_destroy failed with error code: " << result;
+          }
+          if (mesh_marker.points.empty()) {
+            LOG(INFO) << "Empty mesh marker.";
+            continue;
+          }
+          mesh_marker_array.markers.push_back(mesh_marker);
         }
-        if (tango_mesh == nullptr || tango_mesh->num_faces == 0) {
-          LOG(INFO) << "Invalid mesh extracted!";
-          continue;
-        }
-        // Make mesh marker from tango mesh.
-        visualization_msgs::Marker mesh_marker;
-        tango_ros_conversions_helper::toMeshMarker(t3dr_updated_indices->indices[i],
-                                           tango_mesh, time_offset_, &mesh_marker);
-        // Free tango mesh once we are finished with it.
-        Tango3DR_Status result = Tango3DR_Mesh_destroy(tango_mesh);
+        Tango3DR_Status result = Tango3DR_GridIndexArray_destroy(&t3dr_updated_indices);
         if (result != TANGO_3DR_SUCCESS) {
-          LOG(ERROR) << "Tango3DR_Mesh_destroy error: " << result;
+          LOG(ERROR) << "Tango3DR_GridIndexArray_destroy failed with error code: " << result;
         }
-        if (mesh_marker.points.empty()) {
-          LOG(INFO) << "Empty mesh marker.";
-          continue;
+        if (mesh_marker_array.markers.empty()) {
+          LOG(INFO) << "Empty mesh array!";
         }
-        mesh_marker_array.markers.push_back(mesh_marker);
-      }
-      Tango3DR_Status result = Tango3DR_GridIndexArray_destroy(t3dr_updated_indices);
-      if (result != TANGO_3DR_SUCCESS) {
-        LOG(ERROR) << "Tango3DR_GridIndexArray_destroy error: " << result;
+        mesh_marker_publisher_.publish(mesh_marker_array);
       }
-      if (mesh_marker_array.markers.empty()) {
-        LOG(INFO) << "Empty mesh array!";
+      // Publish Tango mesh as occupancy grid.
+      if (occupancy_grid_publisher_.getNumSubscribers() > 0) {
+        Tango3DR_Status result  = Tango3DR_updateFullFloorplan(t3dr_context_);
+        if (result == TANGO_3DR_SUCCESS) {
+          Tango3DR_Vector2 origin;
+          Tango3DR_ImageBuffer image_grid;
+          result = Tango3DR_extractFullFloorplanImage(
+              t3dr_context_, TANGO_3DR_LAYER_OBSTACLES, &origin, &image_grid);
```

Comment:
Yes, in general this is a good idea. But at the moment, I don't see a strong use case for this, so we can keep it as a low prio feature request for now.